transitmatters/t-performance-dash
# V4/date preset logic

![image](https://user-images.githubusercontent.com/46229349/232135198-acef854e-8285-44ce-9a05-7148fedbee32.png)

## Motivation

Our dashboards were set up to remember date params across sections, but not date presets. So if you returned to a section it would always say `custom` as the range.

Also, navigating to a URL with a date param matching a preset will now apply that preset to the date picker.

Lastly, improvements to range/single trip date swap.


## Changes

Adds `datePresetConfig` zustand state management. Keeps track of the preset for date picker across sections.

## Testing Instructions

1. From `overview` page, select preset then navigate to any subpage. Preset will stick.
2. From any `line` subpage, set preset date, navigate to new section, and return. Preset will stick. Same with `trips` section.
3. Navigate via URL to any page with a preset date range, like http://localhost:3000/red/slowzones/?endDate=2022-12-31&startDate=2022-01-01 you will see that the preset is automatically applied.
4. Switch back and forth between single/range tabs on `trips` page. Verify logic is intuitive.
5.  Try anything else that you think should work. Hopefully covered all cases, but please try to find edge cases.
